src/ucs/config/parser.c

```diff
@@ -1512,16 +1525,69 @@ static char *ucs_config_get_value_from_config_file(const char *name)
     return kh_val(&ucs_config_file_vars, iter);
 }
 
+static int ucs_config_parse_check_filter(const char *name, const char *value)
+{
+    typedef struct {
+        const char *name;
+        const char *(*value_f)();
+    } filter_t;
+
+    static filter_t filters[] = {{UCS_CPU_VENDOR_LABEL, ucs_cpu_vendor_name},
+                                 {UCS_CPU_MODEL_LABEL, ucs_cpu_model_name},
+                                 {UCS_SYS_DMI_PRODUCT_NAME_LABEL,
+                                  ucs_sys_dmi_product_name}};
+    filter_t *filter;
+
+    ucs_carray_for_each(filter, filters, ucs_static_array_size(filters)) {
+        if ((strcmp(name, filter->name) == 0) &&
+            (fnmatch(value, filter->value_f(), FNM_CASEFOLD) != 0)) {
+            /**
+             * The value does not match the pattern for this filter. E.g.
+             * configuration file contains the line: CPU model = v1.*, and
+             * ucs_cpu_model_name() retruns "v2.0".
+             */
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+static void
+ucs_config_parse_set_section_info(ucs_config_parse_section_t *section_info,
+                                  const char *section, const char *name,
+                                  const char *value)
+{
+    if (strcmp(section, section_info->name) != 0) {
+        /* A new section has started. Update section information. */
+        ucs_strncpy_zero(section_info->name, section,
+                         sizeof(section_info->name));
+        section_info->skip = 0;
```

Comment:
can we remove this line? section_info->skip is set anyway in line 1571

Reply:
Removed.